### PR TITLE
Added endpoints to serve workflow timing diagrams

### PR DESCRIPTION
**Pull request by cjllanwarne**

### CODE REVIEW

**Comment by geoffjentry** [comment]
Should the diagram generation be part of Cromwell itself, or e.g. part of something like Cromtool?

**Comment by cjllanwarne** [comment]
@geoffjentry this currently hosts an HTML page but pushes all the real work out to the client

**Comment by geoffjentry** [comment]
@cjllanwarne Right, but what I meant was that I'm not sure "give me a diagram of workflow timings" is really something which should be served up at all by Cromwell. Rather give access to the data and let something else figure out what they want to do with it.

**Comment by cjllanwarne** [comment]
@geoffjentry I don't see any harm in having a 2.7K file hosted by the cromwell server for the convenience it brings as an easy and simple way of getting timing information from cromwell. Are you saying it should *also* be part of cromtool?

**Comment by geoffjentry** [comment]
I'm just thinking towards a future where some of these odds and ends move to a more appropriate place (e.g. the highlighter functionality of the jar). If all of the information can be gleaned from the existing metadata endpoint (I think that's true?) then my point is that the fancification can happen elsewhere

**Comment by cjllanwarne** [comment]
That is correct - this already uses the metadata endpoint to get its data. The fancification can indeed happen anywhere - you can download the HTML and put a metadata file next to it and it works just the same

File: src/main/scala/cromwell/webservice/CromwellApiService.scala [hunk 1]
```diff
@@ -169,4 +169,9 @@ trait CromwellApiService extends HttpService with PerRequestCreator {
           complete(StatusCodes.BadRequest, s"Invalid workflow ID: '$workflowId'.")
       }
     }
+
+  def timingRoute =
```
Comment (kshakir) [inline src/main/scala/cromwell/webservice/CromwellApiService.scala]: Add a basic test for your route to at least look for `<html>`.

File: src/main/resources/workflowTimings/workflowTimings.html [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+                chart.draw(graphData, options);
+            });
+        }
+    </script>
+</head>
+<body>
+
+<div id="chart_div"></div>
+</body>
+</html>
```
Comment (kshakir) [inline src/main/resources/workflowTimings/workflowTimings.html]: Curiosity: Why do so many folks hate the terminal newline? Saving this byte causes all sorts of minor pains elsewhere, for example with `cat`.

File: src/main/resources/workflowTimings/workflowTimings.html [hunk 3]
```diff
@@ -0,0 +1,67 @@
+<html>
+<head>
+    <script type="text/javascript" src="https://www.google.com/jsapi"></script>
+    <script src="https://ajax.googleapis.com/ajax/libs/jquery/1.11.3/jquery.min.js"></script>
+    <script type="text/javascript">
+        google.load("visualization", "1.1", {packages:["gantt"]});
+        google.setOnLoadCallback(drawChart);
+
+        function drawChart() {
+            $.getJSON("./metadata", function( data ) {
```
Comment (scottfrazer) [inline src/main/resources/workflowTimings/workflowTimings.html]: Clever, I like it.

**Comment by kshakir** [comment]
Not sure how much UI we do within the cromwell code base. Agree with Jeff that perhaps this might be a better contribution to another UI team, but I'm not the ticket master. Also, I don't really have the ramp up to review HTML/JS for code style conventions / quality myself.

Tried it out locally. As long as the jobs don't run too fast, the endpoint seems to work pretty well!

Btw, when _all_ calls run with start/stop _inside_ a particular second, the page renders with a cryptic red box that says:

> Cannot read property 'v' of undefined

Probably not a big deal unless we expect lots of folks running `ThreeStep.wdl` like I did.

```javascript
// At least one row must span a second 00.

// Ok
var dataRow = ['call_name','call_name', null,
  new Date("2015-11-02T18:00:00.000-05:00"),
  new Date("2015-11-02T18:00:00.001-05:00"),
  null , 100, null];

// or also ok
var dataRow = ['call_name','call_name', null,
  new Date("2015-11-02T18:00:00.999-05:00"),
  new Date("2015-11-02T18:00:01.000-05:00"),
  null , 100, null];

// But if all rows within a second, then error!

// Red box of death
var dataRow = ['call_name','call_name', null,
  new Date("2015-11-02T18:00:00.001-05:00"),
  new Date("2015-11-02T18:00:00.999-05:00"),
  null , 100, null];

// Also an error too
var dataRow = ['call_name','call_name', null,
  new Date("2015-11-02T18:00:00.001-05:00"),
  new Date("2015-11-02T18:00:00.002-05:00"),
  null , 100, null];
```

File: src/main/scala/cromwell/engine/backend/CallMetadata.scala [hunk 4]
```diff
@@ -12,5 +12,6 @@ case class CallMetadata(inputs: Map[String, WdlValue],
                         end: Option[DateTime],
                         jobId: Option[String],
                         returnCode: Option[Int],
+                        shardIndex: Int,
```
Comment (mcovarr) [inline src/main/scala/cromwell/engine/backend/CallMetadata.scala]: good idea!

File: src/main/resources/workflowTimings/workflowTimings.html (hunk 3 above)
Comment (mcovarr) [inline src/main/resources/workflowTimings/workflowTimings.html]: Sneaky.  :smile:

**Pull request merged**